### DISCUSSION TITLE
Changing send ports during loop

### DISCUSSION THREAD

**Discussion by Leandro44** [Arduino-IRremote/Arduino-IRremote#1196] [Q&A] [answered]
I'm having trouble with selecting the pins which I'm using to send the IR commands in my Arduino nano, the code works fine when I fix a pin to send a Samsung signal, but for the application I'm making I need to change the pins dynamically as in the sendNEC command, where I can select the pin I want to send the command, I've tried to set a variable to be the pin as in the documentation but I can't make it work, any suggestions on how can I do that pin selection to send the commands?

**Comment by ArminJo** [comment]
What exactly is not understandable at: https://github.com/Arduino-IRremote/Arduino-IRremote#send-pin

**Reply by Leandro44** [reply]
I'm not the most experienced in programming, I couldn't set the IR_SEND_PIN as a variable, the SEND_PWM_BY_TIMER was on but I reviewed that and still wasn't able to make it work, I might be doing something wrong but I don't know what, I've tried in every way I know and even called help from a more skilled co-worker and neither of us could make it work, would be very helpful if you could send an example of a real code on how to do that if you can.
A suggestion to help future users not as skilled as I am would be to make it like in the TinyIRSender, where you can select the pin more easily as in SendNEC and SendNECMinimal.

I loved your library, it helped me a lot, just this little situation of setting the send pin as a variable I couldn't figure it out.
English isn't my first language, hope that I made myself clear with my questions. Thanks in advance

**Reply by ArminJo** [reply, marked as answer]
I extended SendDemo for you.

**Reply by Leandro44** [reply]
Thanks, I wasn't setting it up properly, with the extension you provided in the SendDemo code worked perfectly for the case I'm using and made me understand what I was doing wrong.
Thanks again, you're amazing!!!
